File: InterviewQuestions/Kardium/chain-printer-1.0-SNAPSHOT-project/chain-printer/src/main/java/com/kardium/exams/printer/LoggingChainPrinterDriver.java
package com.kardium.exams.printer;

/*
 * Copyright Kardium Inc. 2023.
 */

/**
 * Logs printer driver calls to standard output.
 */
public class LoggingChainPrinterDriver implements ChainPrinterDriver {
    @Override
    public void fire(int solenoidIndex) {
        System.out.println("Fire solenoid " + solenoidIndex);
    }

    @Override
    public void step() {
        System.out.println("Step chain 1 mm");
    }

    @Override
    public void linefeed() {
        System.out.println("Linefeed");
    }
}

File: InterviewQuestions/Kardium/chain-printer-1.0-SNAPSHOT-project/ValentineSsebuyungoSubmission/LoggingChainPrinterDriver.cpp
#include "LoggingChainPrinterDriver.h"

void LoggingChainPrinterDriver::fire(int solenoidIndex) {
    std::cout << "Fire solenoid " << solenoidIndex << std::endl;
}

void LoggingChainPrinterDriver::step() {
    std::cout << "Step chain 1 mm" << std::endl;
}

void LoggingChainPrinterDriver::linefeed() {
    std::cout << "Linefeed" << std::endl;
}
